diff --git a/node_modules/builder-util/out/util.js b/node_modules/builder-util/out/util.js
index 8064112..b6e3d81 100644
--- a/node_modules/builder-util/out/util.js
+++ b/node_modules/builder-util/out/util.js
@@ -505,7 +505,7 @@ function isEmptyOrSpaces(s) {
 }
 
 function isTokenCharValid(token) {
-  return /^[\w\/=+-]+$/.test(token);
+  return /^[.\w\/=+-]+$/.test(token);
 }
 
 function addValue(map, key, value) {
