diff --git a/node_modules/electron-spellchecker/src/context-menu-builder.js b/node_modules/electron-spellchecker/src/context-menu-builder.js
index afded5f..6d647af 100644
--- a/node_modules/electron-spellchecker/src/context-menu-builder.js
+++ b/node_modules/electron-spellchecker/src/context-menu-builder.js
@@ -8,7 +8,7 @@ let d = require('debug')('electron-spellchecker:context-menu-builder');
 const contextMenuStringTable = {
   copyMail: () => `Copy Email Address`,
   copyLinkUrl: () => `Copy Link`,
-  openLinkUrl: () => `Open Link`,
+  openLinkUrl: () => `Open Link in Browser`,
   copyImageUrl: () => `Copy Image URL`,
   copyImage: () => `Copy Image`,
   addToDictionary: () => `Add to Dictionary`,
