diff --git a/dist/config.js b/dist/config.js
index 4932902ac605eaa2c6d134fbcfdd7aae23d6accb..c3139159b4545fcc261b42afd84edd7a75a1368b 100644
--- a/dist/config.js
+++ b/dist/config.js
@@ -10,7 +10,7 @@ const __dirname = path.dirname(fileURLToPath(import.meta.url));
 const env = envVar.from(process.env);
 export const llamaDirectory = path.join(__dirname, "..", "llama");
 export const llamaToolchainsDirectory = path.join(llamaDirectory, "toolchains");
-export const llamaPrebuiltBinsDirectory = path.join(__dirname, "..", "llamaBins");
+export const llamaPrebuiltBinsDirectory = global.LLAMA_PREBUILT_BINS_DIRECTORY || path.join(__dirname, "..", "llamaBins");
 export const llamaLocalBuildBinsDirectory = path.join(llamaDirectory, "localBuilds");
 export const llamaBinsGrammarsDirectory = path.join(__dirname, "..", "llama", "grammars");
 export const llamaCppDirectory = path.join(llamaDirectory, "llama.cpp");
@@ -27,7 +27,7 @@ export const localXpacksCacheDirectory = path.join(xpackDirectory, "cache");
 export const buildMetadataFileName = "_nlcBuildMetadata.json";
 export const xpmVersion = "^0.16.3";
 export const builtinLlamaCppGitHubRepo = "ggerganov/llama.cpp";
-export const builtinLlamaCppRelease = await getBinariesGithubRelease();
+export const builtinLlamaCppRelease = "b2608";
 export const isCI = env.get("CI")
     .default("false")
     .asBool();
