diff --git a/lib/ComponentLoader.d.ts b/lib/ComponentLoader.d.ts
index 8e7731dbdaa53005d58d55e6a834d20d032eff7e..faa04bafb5667a24ceabbf0432439e9b42856f4b 100644
--- a/lib/ComponentLoader.d.ts
+++ b/lib/ComponentLoader.d.ts
@@ -79,7 +79,7 @@ export class ComponentLoader {
      * @param {import("fbp-graph/lib/Types").GraphNodeMetadata} metadata
      * @returns {Promise<import("./Component").Component>}
      */
-    protected createComponent(name: string, component: ComponentDefinitionWithoutGraph, metadata: import("fbp-graph/lib/Types").GraphNodeMetadata): Promise<import("./Component").Component>;
+    createComponent(name: string, component: ComponentDefinitionWithoutGraph, metadata: import("fbp-graph/lib/Types").GraphNodeMetadata): Promise<import("./Component").Component>;
     /**
      * @param {import("fbp-graph").Graph|object|string} cPath
      * @returns {boolean}
diff --git a/src/lib/ComponentLoader.js b/src/lib/ComponentLoader.js
index b3215074ff63d5be2f352a4dc94635f82212836c..98c7b653da5ec47072656afaef1c50b6c7ebd646 100644
--- a/src/lib/ComponentLoader.js
+++ b/src/lib/ComponentLoader.js
@@ -226,9 +226,8 @@ export class ComponentLoader {
     return promise;
   }
 
-  // Creates an instance of a component.
   /**
-   * @protected
+   * Creates an instance of a component.
    * @param {string} name
    * @param {ComponentDefinitionWithoutGraph} component
    * @param {import("fbp-graph/lib/Types").GraphNodeMetadata} metadata